engine/crates/engine-v2/src/engine/cache.rs

```diff
@@ -56,21 +49,24 @@ impl std::fmt::Display for Key<'_> {
                 // operation name.
                 hasher.update(&[0x00]);
                 match document {
-                    Document::PersistedQueryExt(ext) => {
+                    Document::AutomaticallyPersistedQuery(ext) => {
                         hasher.update(b"apq");
                         hasher.update(&[0x00]);
                         hasher.update(&ext.version.to_ne_bytes());
                         hasher.update(&ext.sha256_hash);
                     }
-                    Document::Id(doc_id) => {
+                    Document::TrustedDocumentId { client_name, doc_id } => {
```

Comment:
:bug: **Bug**
The change from `Document::Id(doc_id)` to `Document::TrustedDocumentId { client_name, doc_id }` in the hashing logic introduces a potential bug. The `client_name` is being hashed, but the length of `client_name` is not being checked for overflow or validity. This could lead to unexpected behavior if `client_name` is excessively long or contains invalid characters. It's important to ensure that the length of `client_name` is within expected bounds and that it contains valid characters before hashing.

```suggestion
                    Document::TrustedDocumentId { client_name, doc_id } => {
                        if client_name.len() > MAX_CLIENT_NAME_LENGTH {
                            return Err(std::fmt::Error);
                        }
                        hasher.update(b"docid");
                        hasher.update(&[0x00]);
                        hasher.update(&client_name.len().to_ne_bytes());
                        hasher.update(client_name.as_bytes());
                        hasher.update(&doc_id.len().to_ne_bytes());
                        hasher.update(doc_id.as_bytes());
                    }
```